backend/controllers/serviceexportrequest/serviceexportrequest_reconcile.go

```diff
@@ -37,33 +42,143 @@ import (
 type reconciler struct {
 	informerScope          kubebindv1alpha2.InformerScope
 	clusterScopedIsolation kubebindv1alpha2.Isolation
+	schemaSource           string
+
+	getBoundSchema    func(ctx context.Context, cache cache.Cache, namespace, name string) (*kubebindv1alpha2.BoundSchema, error)
+	createBoundSchema func(ctx context.Context, cl client.Client, schema *kubebindv1alpha2.BoundSchema) error
 
-	getAPIResourceSchema       func(ctx context.Context, cache cache.Cache, name string) (*kubebindv1alpha2.APIResourceSchema, error)
 	getServiceExport           func(ctx context.Context, cache cache.Cache, ns, name string) (*kubebindv1alpha2.APIServiceExport, error)
 	createServiceExport        func(ctx context.Context, cl client.Client, resource *kubebindv1alpha2.APIServiceExport) error
-	createAPIResourceSchema    func(ctx context.Context, cl client.Client, schema *kubebindv1alpha2.APIResourceSchema) (*kubebindv1alpha2.APIResourceSchema, error)
 	deleteServiceExportRequest func(ctx context.Context, cl client.Client, namespace, name string) error
 }
 
 func (r *reconciler) reconcile(ctx context.Context, cl client.Client, cache cache.Cache, req *kubebindv1alpha2.APIServiceExportRequest) error {
-	var errs []error
+	if err := r.ensureBoundSchemas(ctx, cl, cache, req); err != nil {
+		conditions.SetSummary(req)
+		return err
+	}
 
 	if err := r.ensureExports(ctx, cl, cache, req); err != nil {
-		errs = append(errs, err)
+		conditions.SetSummary(req)
+		return err
 	}
 
+	// TODO(mjudeikis): we could potentially add finallizer to APIServiceExport above or "adopt" boundschemas
+	// with owner references once export is created.
+	// https://github.com/kube-bind/kube-bind/issues/297
+
 	conditions.SetSummary(req)
 
-	return utilerrors.NewAggregate(errs)
+	return nil
+}
+
+func (r *reconciler) ensureBoundSchemas(ctx context.Context, cl client.Client, cache cache.Cache, req *kubebindv1alpha2.APIServiceExportRequest) error {
+	// Ensure all bound schemas exist
+	for _, res := range req.Spec.Resources {
+		parts := strings.SplitN(r.schemaSource, ".", 3)
+		if len(parts) != 3 { // We check this in validation, but just in case.
+			return fmt.Errorf("invalid schema source: %q", r.schemaSource)
```

Comment:
Nit: In absence of an `err`, "malformed" might better convey what the issue is with the schema source.